File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnRhythmFilter.cs

namespace GracenoteSDK {

/**************************************************************************
** GnRhythmFilter class
*/
public class GnRhythmFilter : global::System.IDisposable {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal GnRhythmFilter(global::System.IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnRhythmFilter obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnRhythmFilter() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnRhythmFilter(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
    }
  }

  public GnRhythmFilter() : this(gnsdk_csharp_marshalPINVOKE.new_GnRhythmFilter(), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  Set option using list element handle
*  @param list_element_handle   [in] element handle
*/
  public void Custom(GnListElement listElement) {
    gnsdk_csharp_marshalPINVOKE.GnRhythmFilter_Custom(swigCPtr, GnListElement.getCPtr(listElement));
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

}

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnRhythmFilter.java

package com.gracenote.gnsdk;

/************************************************************************** 
** {@link GnRhythmFilter} class 
*/ 
 
public class GnRhythmFilter {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected GnRhythmFilter(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnRhythmFilter obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnRhythmFilter(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

  public GnRhythmFilter() throws com.gracenote.gnsdk.GnException {
    this(gnsdk_javaJNI.new_GnRhythmFilter(), true);
  }

/** 
*  Set option using list element handle 
*  @param list_element_handle   [in] element handle 
*/ 
 
  public void custom(GnListElement listElement) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnRhythmFilter_custom(swigCPtr, this, GnListElement.getCPtr(listElement), listElement);
  }

}
